src/libre/ast.h

```diff
@@ -98,6 +98,7 @@ enum ast_flags {
 	AST_FLAG_ANCHORED_START  = 1 << 6,
 	AST_FLAG_ANCHORED_END    = 1 << 7,
 	AST_FLAG_END_NL          = 1 << 8,
+	AST_FLAG_MATCHES_1NEWLINE= 1 << 9,
```

Comment:
lol

Reply:
There will be even more niche start/end anchor flags coming soon. Expressing PCRE's anchoring rules statically gets complicated.